mobius-core/src/test/java/com/spotify/mobius/MobiusLoopControllerTest.java

```diff
@@ -628,23 +631,36 @@ public void shouldLogBeforeAndAfterUpdate() {
   private static class KnownThreadWorkRunner implements WorkRunner {
     private final ExecutorService executorService = Executors.newSingleThreadExecutor();
     private volatile Thread workerThread = null;
+    @Nonnull private final Lock lock = new ReentrantLock();
```

Comment:
This might in fact now be more easily implemented in terms of an `ExecutorServiceWorkRunner`: 

 ```
  private final ExecutorServiceWorkRunner delegate = ESWR(newSingleThreadExecutor());

public void post(final Runnable runnable) {
   delegate.post({
      workerThread = Thread.currentThread();
      runnable.run();
  }
}
```